pip/pip-425.md

```diff
@@ -0,0 +1,127 @@
+
+For removed addresses, the quarantine duration will increase exponentially. The quarantine duration can be customized via new optional configurations `serviceUrlInitFaultQuarantineDuration` and `serviceUrlMaxFaultQuarantineDuration` in `ClientBuilder`
+That is to say, assuming that `serviceUrlInitFaultQuarantineDuration` is 1 minute and `serviceUrlMaxFaultQuarantineDuration` is 30 minutes, after host1 is judged as unavailable for the first time, it will be isolated for 1 minute and then added to `availableAddressPool` again. If it is judged as unavailable again the next time, it will be isolated for 2 minutes. Similarly, if it is judged as unavailable continuously, the isolation time will be 4 minutes, 8 minutes, 16 minutes, 30 minutes (it reaches serviceUrlMaxFaultQuarantineDuration and no longer increases), 30 minutes... If host1 is determined to be available, the next quarantine duration will be reset to 1 minute.
+
+
+### Usage Example
+```java
+    private void example() throws PulsarClientException {
+        // The failed endpoint will be removed and retried after a period of time, with an initial quarantine duration of 30 seconds increasing exponentially. After reaching the maximum quarantine duration of 1 hour, will be maintained once an hour until a certain succeeds, and the quarantine duration is reset to 0.
+        PulsarClient pulsarClient = PulsarClient.builder()
+                .serviceUrl("pulsar://localhost1:6650,localhost2:6650,localhost3:6650")
+                .serviceUrlInitFaultQuarantineDuration(30000, TimeUnit.MILLISECONDS)
+                .serviceUrlMaxFaultQuarantineDuration(1, TimeUnit.HOURS)
+                .build();    
+
+        // disable removing unavailable serviceUrl endpoints by pass 0 to serviceUrlInitFaultQuarantineDuration and serviceUrlMaxFaultQuarantineDuration
+        PulsarClient pulsarClientDisableReovery = PulsarClient.builder()
+                .serviceUrl("pulsar://localhost1:6650,localhost2:6650,localhost3:6650")
+                .serviceUrlInitFaultQuarantineDuration(0, TimeUnit.MILLISECONDS)
+                .serviceUrlMaxFaultQuarantineDuration(0, TimeUnit.MILLISECONDS)
+                .build();                
+}
+
+```
+## Public-facing Changes
+
+### Public API
+
+1. Add two optional configs `serviceUrlInitFaultQuarantineDuration` and `serviceUrlMaxFaultQuarantineDuration` in `ClientBuilder`
+```java
+    /**
+     * Configure the service URL init fault quarantine duration.
+     *
+     * <p>When the client is unable to connect to the service URL, it will wait for a certain amount of time before
+     * trying to recover the service URL endpoint. The init quarantine duration can be configured using this method.
```

Comment:
this description isn't correct. There's not recovery happening. It's simply about quarantining the endpoint url for a specific duration that is determined in a certain emponential way. A successful usage of the endpoint url will reset the duration to the initial value and move it back to the available addresses pool.